[14.0] [IMP] shopinvader: Make SESS_CART_ID optional when a partner is logged in

paradoxxxzero opened the pull request:
This PR add a way to get the current "active" partner cart without session cart id.
This PR also add a `restrict_cart_to_partner` option on backend to restrict cart access to its logged in partner owner.

hparfr reviewed (approved):
LGTM

paradoxxxzero commented:
Is there anything still blocking? (FYI @hparfr )

mourad-ehm commented on shopinvader/tests/test_cart.py:
```diff
@@ -440,6 +440,32 @@ def test_cart_misc_data_update(self):
+
+        # Is it necessary to create a cart in this case?
+        # This is done for backward compatibility for now
+        self.assertNotEqual(anon_cart.id, cart["id"])
```
Isn't this test missing (restrict_cart_to_partner)?  self..assertEqual(anon_cart.partner_id, self.partner)

paradoxxxzero committed:
[IMP] shopinvader: Make SESS_CART_ID optional when a partner is logged in
Also add restrict_cart_to_partner option on backend to prevent access to other partner carts.
